subiquity/common/types.py

```diff
@@ -333,11 +333,69 @@ class StorageResponseV2:
 
 @attr.s(auto_attribs=True)
 class GuidedResizeValues:
+    install_max: int
     minimum: int
     recommended: int
     maximum: int
 
 
+@attr.s(auto_attribs=True)
+class GuidedStorageTargetReformat:
+    disk_id: str
+
+
+@attr.s(auto_attribs=True)
+class GuidedStorageTargetResize:
+    disk_id: str
+    partition_number: int
+    new_size: int
+    minimum: Optional[int]
+    recommended: Optional[int]
+    maximum: Optional[int]
+
+    @staticmethod
+    def from_recommendations(part, resize_vals):
+        return GuidedStorageTargetResize(
+                disk_id=part.device.id,
+                partition_number=part.number,
+                new_size=resize_vals.recommended,
+                minimum=resize_vals.minimum,
+                recommended=resize_vals.recommended,
+                maximum=resize_vals.maximum,
+                )
+
+
+@attr.s(auto_attribs=True)
+class GuidedStorageTargetUseGap:
+    disk_id: str
+    gap: Gap
+
+
+@attr.s(auto_attribs=True)
+class GuidedChoiceV2:
+    target: Union[GuidedStorageTargetReformat,
+                  GuidedStorageTargetResize,
+                  GuidedStorageTargetUseGap]
```

Comment:
I think I had expected that Union[GuidedStorageTargetReformat, GuidedStorageTargetResize,  GuidedStorageTargetUseGap] would be called GuidedStorageTarget, but I guess it's not really important.

Reply:
Done, see new commits